homeassistant/components/tuya/switch.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+"""Support for Tuya switches."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from tuya_iot import TuyaDevice, TuyaDeviceManager
+
+from homeassistant.components.switch import DOMAIN as DEVICE_DOMAIN, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .base import TuyaHaDevice
+from .const import (
+    DOMAIN,
+    TUYA_DEVICE_MANAGER,
+    TUYA_DISCOVERY_NEW,
+    TUYA_HA_DEVICES,
+    TUYA_HA_TUYA_MAP,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+TUYA_SUPPORT_TYPE = {
+    "kg",  # Switch
+    "cz",  # Socket
+    "pc",  # Power Strip
+    "bh",  # Smart Kettle
+    "dlq",  # Breaker
+    "cwysj",  # Pet Water Feeder
+    "kj",  # Air Purifier
+    "xxj",  # Diffuser
+}
+
+# Switch(kg), Socket(cz), Power Strip(pc)
+# https://developer.tuya.com/en/docs/iot/categorykgczpc?id=Kaiuz08zj1l4y
+DPCODE_SWITCH = "switch"
+
+# Air Purifier
+# https://developer.tuya.com/en/docs/iot/categorykj?id=Kaiuz1atqo5l7
+# Pet Water Feeder
+# https://developer.tuya.com/en/docs/iot/f?id=K9gf46aewxem5
+DPCODE_ANION = "anion"  # Air Purifier - Ionizer unit
+# Air Purifier - Filter cartridge resetting; Pet Water Feeder - Filter cartridge resetting
+DPCODE_FRESET = "filter_reset"
+DPCODE_LIGHT = "light"  # Air Purifier - Light
+DPCODE_LOCK = "lock"  # Air Purifier - Child lock
+# Air Purifier - UV sterilization; Pet Water Feeder - UV sterilization
+DPCODE_UV = "uv"
+DPCODE_WET = "wet"  # Air Purifier - Humidification unit
+DPCODE_PRESET = "pump_reset"  # Pet Water Feeder - Water pump resetting
+DPCODE_WRESET = "water_reset"  # Pet Water Feeder - Resetting of water usage days
+
+
+DPCODE_START = "start"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, _entry: ConfigEntry, async_add_entities
+):
+    """Set up tuya sensors dynamically through tuya discovery."""
+    _LOGGER.debug("switch init")
+
+    hass.data[DOMAIN][TUYA_HA_TUYA_MAP].update({DEVICE_DOMAIN: TUYA_SUPPORT_TYPE})
```

Comment:
You can just assign a single key instead of using update.
```suggestion
    hass.data[DOMAIN][TUYA_HA_TUYA_MAP][DEVICE_DOMAIN] = TUYA_SUPPORT_TYPE
```

Also please update this constant to be indexed by config entry. `hass.data[DOMAIN][_entry.entry_id][TUYA_HA_TUYA_MAP]`

Reply:
fixed